### PR TITLE
Added userinfo endpoint

### LINKED ISSUE #289: How to get the userinfo
Hi,

I'm wondering is it possible to expose the user profile endpoint? So I can pass the information to my app. 

This PR(https://github.com/bitly/oauth2_proxy/pull/388) seems do the trick!?

thanks

### PR DESCRIPTION
**Pull request by kbabuadze**
## Description
Added /oauth2/userinfo endpoint which outputs email  from the Authenticated session in JSON format.

## Motivation and Context

Fixes #289 

## How Has This Been Tested?

Tested the change with authenticated and non-authenticated user. Oauth provider Github.

## Checklist:

I've updated documentation , please let me know if CHANGELOG should be updated by me.

- [X] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -585,6 +590,24 @@ func (p *OAuthProxy) SignIn(rw http.ResponseWriter, req *http.Request) {
...
+
+	session, err := p.getAuthenticatedSession(rw, req)
+	if err != nil {
+		http.Error(rw, "unauthorized request", http.StatusUnauthorized)
+		return
+	}
+	userInfo.Email = session.Email
+	userJSON, _ := json.Marshal(userInfo)
+
+	rw.Write(userJSON)
```
[the remark this answers is not in the document]
Comment (kbabuadze) [reply oauthproxy.go]: Indeed that must be much efficient. Thanks for the suggestion.

File: oauthproxy.go [hunk 2]
```diff
@@ -585,6 +590,24 @@ func (p *OAuthProxy) SignIn(rw http.ResponseWriter, req *http.Request) {
...
+	var userInfo struct {
+		Email string `json:"email"`
+	}
+
+	session, err := p.getAuthenticatedSession(rw, req)
+	if err != nil {
+		http.Error(rw, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+	userInfo.Email = session.Email
```
Comment (syscll) [inline oauthproxy.go]: Sorry for the nit, but you could remove lines 595-599 and do this:
```suggestion
	userInfo := struct {
		Email string `json:"email"`
	}{session.Email}
```

Comment (kbabuadze) [reply oauthproxy.go]: No sorry , on the contrary, I am open to take suggestion from an experienced person. Thanks.

**Comment by syscll** [comment]
Sorry for the back and fourth. The code LGTM :+1: 

Would it be possible to add a couple of test cases? Then I think we're good to get this merged.

**Comment by kbabuadze** [comment]
Sure! I'll work on that and get back to you once it's done.

**Comment by syscll** [comment]
@kbabuadze awesome!

Don't forget to add an entry to the CHANGELOG, too.

**Comment by kbabuadze** [comment]
Thank you ! :tada:

**Comment by syscll** [comment]
@JoelSpeed can you take a look please

**Pull request merged**